v2/client.go

```diff
@@ -139,18 +146,29 @@ func requestContentType(ct string) requestOption {
 	}
 }
 
-// buildURL builds a url to /api/v2/... using the given pathElements. It
-// url escapes each path element, so the caller doesn't need to worry about
-// that.
+// buildURL builds a url to /api/v2/... using the given pathElements.
+// It url escapes each path element, so the caller doesn't need to worry about that.
 func (c *Client) buildURL(pathElements ...any) *url.URL {
 	elem := make([]string, 1, len(pathElements)+1)
 	elem[0] = "/api/v2"
 	for _, pathElement := range pathElements {
-		elem = append(elem, fmt.Sprint(pathElement))
+		elem = append(elem, url.PathEscape(fmt.Sprint(pathElement)))
```

Comment:
Another note for future us: will likely want to use [url.QueryEscape](https://pkg.go.dev/net/url#QueryEscape) for any query params we allow interacting with on endpoints that accept them